src/Ractive/prototype/shared/hooks/Hook.js

```diff
@@ -32,7 +32,12 @@ export default class Hook {
 	fire ( ractive, arg ) {
 		function call ( method ) {
 			if ( ractive[ method ] ) {
-				arg ? ractive[ method ]( arg ) : ractive[ method ]();
+				try {
+          arg ? ractive[ method ]( arg ) : ractive[ method ]();
+        } catch (e) {
+          ractive.fire('error', e);
+          console.error(e);
```

Comment:
I would use logging from src/util/log. Perhaps an error method is in order. I think it may be safer to let it die here than possibly get in an undefined state further on, though I'm often wrong.